x-pack/plugins/cases/server/client/cases/similar.ts

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { fromKueryExpression, nodeBuilder } from '@kbn/es-query';
+import type { SimilarCasesSearchRequest } from '../../../common/types/api';
+import {
+  type CasesFindResponse,
+  CasesFindResponseRt,
+  SimilarCasesSearchRequestRt,
+} from '../../../common/types/api';
+import { decodeWithExcessOrThrow, decodeOrThrow } from '../../common/runtime_types';
+
+import { createCaseError } from '../../common/error';
+import type { CasesClientArgs } from '..';
+import { flattenCaseSavedObject } from '../../common/utils';
+
+/**
+ * Retrieves cases similar to a given Case
+ *
+ * @ignore
+ */
+export const similar = async (
+  params: SimilarCasesSearchRequest,
+  clientArgs: CasesClientArgs
+): Promise<CasesFindResponse> => {
+  const {
+    services: { caseService },
+    logger,
+  } = clientArgs;
+
+  try {
+    const paramArgs = decodeWithExcessOrThrow(SimilarCasesSearchRequestRt)(params);
+
+    const filterExpressions = Object.keys(paramArgs.observables).flatMap((typeKey) => {
+      return Object.values(paramArgs.observables[typeKey]).map((observableValue) => {
+        return fromKueryExpression(
+          `cases.attributes.observables:{value: ${observableValue} AND typeKey: ${typeKey}}`
```

Comment:
the custom bit, now that I am thinking about it we should escape the value maybe :D